### PR TITLE
Migrate docs site to Vercel

### PR DESCRIPTION
**Pull request by delucis**
#### What kind of changes does this PR include?

- Something else!

#### Description

- [x] Migrates the Starlight docs to deploy to Vercel
- [x] Disables our `lastUpdated` feature, which doesn’t currently support Vercel due to them blocking deep clones
- [x] Adds a `vercel.json` with redirect config previously provided by `_redirects` for Netlify
- [x] Removes old Netlify config files. (Our `_headers` file was setting cache headers that are the default for Astro projects on Vercel.)
- [ ] Will need to updated DNS after this merges

### CODE REVIEW

**Comment by delucis** [comment]
Haven’t figured out the multilingual 404 support just yet. On Netlify this was pretty straightforward in `_redirects`:
```
/:lang/*    /:lang/404/   404
```

**Comment by HiDeoo** [comment]
> Haven’t figured out the multilingual 404 support just yet.

Could this work like [rewrites with regex path matching](https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites#regex-path-matching) in a Next config where to match a regex path you have wrap the regex in parenthesis *after a parameter*? E.g. `"source": "/:lang/:useless(.*)/"`?

Altho I guess this would end up being the same as `:useless*` 🤔

**Comment by delucis** [comment]
Yeah that’s what I tried (thanks to @tony-sull’s guidance):
```js
  "redirects": [
    {
      "source": "/:lang/(.*)/",
      "destination": "/:lang/404/",
      "statusCode": 404
    }
  ],
```
The above doesn’t seem to do anything but when I tried `"source": "/:lang/(.*)"` (no trailing slash) it seemed to intercept requests greedily and break asset loading: https://starlight-2dm4ui5ee-astrodotbuild.vercel.app/

**Comment by delucis** [comment]
Closing until we can figure out the 404 routing.

File: docs/vercel.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    { "src": "/(ph$|ph/)(.*)", "dest": "https://astro-houston-ph.pages.dev/ph/$2" },
+
+    { "src": "(.*)/([^./]+)$", "dest": "$1/$2/", "status": 301 },
+    { "src": "(.*)/index.html$", "dest": "$1/", "status": 301 },
+
+    { "handle": "filesystem" },
+
+    { "src": "/zh/(.*)", "dest": "/zh-cn/$1", "status": 301 },
+
+    { "src": "/(?<lang>[^/]*)/(.*)", "dest": "/$lang/404/", "status": 404 }
```
Comment (delucis) [inline docs/vercel.json:20]: Support multilingual 404s.

#### Tests
- https://starlight-i3v427u6r-astrodotbuild.vercel.app/undefined
- https://starlight-i3v427u6r-astrodotbuild.vercel.app/pt-br/undefined

**Review by tony-sull — APPROVED** [review]
This looks great!

Before merging, make sure to disconnect the Netlify builds from the github repo just in case.  That will freeze Netlify on the last non-Vercel build and keep it alive until after the domain/DNS settings in Vercel take over

**Pull request merged**